src/nvim/terminal.c

```diff
@@ -288,8 +288,9 @@ void terminal_close(Terminal *term, char *msg)
 
   term->forward_mouse = false;
   term->closed = true;
+  buf_T *buf = handle_get_buffer(term->buf_handle);
```

Comment:
Okay, so I did some tests.

1. `handle_get_buffer()` returns `NULL` for every invalid handle, so that first `if` isn't needed.
2. `terminal_close()` only gets called from 2 locations, `on_process_exit()` and `close_buffer()`. (The comment doesn't indicate that `terminal_close()` might be called again.) I jumped back and forth trough the code and can't see how the `apply_autocmds()` could ever be called with `buf == NULL`, but well, it doesn't hurt either. :)